pkg/squashfs/squashfs.go

```diff
@@ -55,23 +73,31 @@ func (s *Snap) ControlMember(controlMember string) ([]byte, error) {
 	return s.ReadFile(filepath.Join("DEBIAN", controlMember))
 }
 
-// MetaMember extracts from meta/ - COMPAT
+// MetaMember extracts from meta/. - COMPAT
 func (s *Snap) MetaMember(metaMember string) ([]byte, error) {
 	return s.ReadFile(filepath.Join("meta", metaMember))
 }
 
-// ExtractHashes does notthing for squashfs snaps - COMAPT
+// ExtractHashes does notthing for squashfs snaps. - COMAPT
 func (s *Snap) ExtractHashes(dir string) error {
 	return nil
 }
 
-// UnpackWithDropPrivs unpacks the meta and puts stuff in place - COMAPT
+// UnpackWithDropPrivs just copies the blob into place. - COMPAT
 func (s *Snap) UnpackWithDropPrivs(instDir, rootdir string) error {
-	// FIXME: actually drop privs
-	return s.Unpack("*", instDir)
+
+	// ensure mount-point and blob dir.
+	for _, dir := range []string{instDir, dirs.SnapBlobDir} {
+		if err := os.MkdirAll(dir, 0755); err != nil {
+			return err
+		}
+	}
+
+	// FIXME: helpers.CopyFile() has no preserve attribute flag yet
+	return runCommand("cp", "-a", s.path, BlobPath(instDir))
 }
 
-// UnpackMeta unpacks just the meta/* directory of the given snap
+// UnpackMeta unpacks just the meta/* directory of the given snap.
 func (s *Snap) UnpackMeta(dst string) error {
```

Comment:
Excellent point, I removed unpackmeta now. As for unpack vs unpackall> we will need unpack() for the kernel snap, it will have to run something like unpack("/path/to/kernel", /boot/grub/ubuntu/ubuntu-kernel_$ver"). We do not need unpackall for squashfs (and its not there unles I'm missing something).